cpp/src/arrow/ipc/json-internal.cc

```diff
@@ -558,7 +558,11 @@ class ArrayWriter {
   Status Visit(const StructArray& array) {
     WriteValidityField(array);
     const auto& type = static_cast<const StructType&>(*array.type());
-    return WriteChildren(type.children(), array.fields());
+    std::vector<std::shared_ptr<Array>> children;
```

Comment:
`StructArray` is no longer storing the `arrow::Array` instances internally; they are constructed on demand by `arr->field(i)`. I am not sure we need to expose the vector of shared_ptr even if it existed